### ISSUE TITLE
Add a plugin project template

### ISSUE THREAD

**Issue by FedericoCeratto** [getpelican/pelican-plugins#436]
To make writing plugins simpler and more streamlined a template could be provided.
This would also help reducing the fragmentation of the documentation.

It could contain a set of example files and directories:

my_plugin (directory)
-  __init__.py  - with a simple import line
- my_plugin.py - a working example including signals, a generator, a reader based on the links below
- test_my_plugin.py - a working unit-test, using  py.test
- Readme.rst - installation instructions, debugging tips
- setup.py - configured with the classifier and metadata from #39
- LICENSE - AGPLv3 license
- HACKING - brief instructions on how to release the plugin

https://www.heroicdebugging.biz/2015/01/16/writing-pelican-plugin-tests/
http://docs.getpelican.com/en/latest/plugins.html#how-to-create-plugins
http://docs.getpelican.com/en/latest/internals.html
https://github.com/getpelican/pelican-plugins/blob/master/Contributing.rst

**Comment by nebulousdog** [comment]
I think this is a great idea. Hoping can pair such with comments over here https://github.com/getpelican/pelican-plugins/issues/425.

**Comment by Lucas-C** [comment]
Sounds like a great idea to me too.

**Comment by pauloxnet** [comment]
Great idea, another important information to write or update a pelican plugin can be code contribution style (es: common settings for flake8, isort, black ? )

**Comment by nebulousdog** [comment]
hey again! I wrote a draft Pelican Plugin template. I included the majority of the things @FedericoCeratto originally proposed such as a `setup.py` config, tests, instructions, and document references, and @pauloxnet's idea about codestyle. There are also some [deploy scripts](https://github.com/nebulousdog/thedropin/blob/master/Pipfile#L20-L30) included for convenience.

see 🛹 https://github.com/nebulousdog/thedropin

I focused on the infrastructure bits, and tried to keep the instructions to the bare minimum required to get someone up and running as fast as possible.

I'd love to hear from all of you about best practices to be sure to include now that the shell is set up.
